### PR TITLE
[WFLY-14087]: Updating JDBC journal documentation

### PR DESCRIPTION
**Pull request by ehsavoie**
* Updating the documentation

Jira: https://issues.redhat.com/browse/WFLY-14087

### CODE REVIEW

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (marekkopecky) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:26]: It seems that mentioned commands are stored in `modules/system/layers/base/org/apache/activemq/artemis/main/artemis-jdbc-store-${ARTEMIS_VERSION}.jar/journal-sql.properties`. But this PR removes details about journal-sql.properties file. Is this change related to https://issues.redhat.com/browse/WFLY-14087 jira and related pool configuration?

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (marekkopecky) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:45]: > The size of the pool should be defined

AFAIU the size is dynamic and can't be hard-coded in some configuration. But user could define is max-size of the pool. Right?

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (ehsavoie) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:26]: Given that this file is provided by Artemis now and is not meant to be overriden by WildFly, I'd rather remove that incorrect documentation

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (marekkopecky) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:45]: > according to the amount of concurrent threads

Can these amount of threads be read somewhere? (eg `/subsystem=ee/managed-thread-factory=...:read-resource()`?) Can you please specify some rules for the size of pool based on the amount of concurrent threads?

Comment (ehsavoie) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:45]: Correct, you can configure a min size and a max size. Also the timeout to get a connection, or the time to return the connection to the pool or renew it. It seems to me that this is more a datasource configuration documentation than one that is proper to the journal

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (marekkopecky) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:26]: Sure, current documentation is incorrect. But new documentation suggest that users should check artemis-jdbc-store jar for SQL commands, but doesn't provide details where these commands could be found in .jar file.

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (marekkopecky) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:45]: Sure, details about datasource commands should be mentioned in datasource chaptes in upstream doc. But so far `The size of the pool should be defined` text in this messaging guides seems to be wrong based on your previous message. Also rules based on messaging streaming operations are still missing here (see https://github.com/wildfly/wildfly/pull/13965/#discussion_r572023909 )

**Review by marekkopecky — CHANGES_REQUESTED** [review]
Some changes needs to be done, see my previous comments.

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:48]: s/taylor/tailor 

unless you mean to ask Andy to do it ;)

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:45]: is that final '+' cruft?

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:46]: s/one of the fact/one fact/g

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (ehsavoie) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:45]: no I wanted a line break

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (marekkopecky) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc]: @ehsavoie This is still not fixed

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (marekkopecky) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:45]: It's not clear where this attribute could be found from the documentation. `/subsystem=datasources/data-source=NAME`?

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (marekkopecky) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:47]: As I mentioned earlier (and as you confirmed), size means "max-pool-size", but the doc doesn't explain it.

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (marekkopecky) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:48]: Ok, there is no 1-1 relation, but we should elaborate more about it. Documentation should contains details about options that we have (link to datasource doc could be used, if something in datasource doc is missing, it could be added in this PR). Documentation should contains recommendation and tips how to set datasource settings correctly.

`You really need to tailor your configuration according to your needs`

This is obvious fact, but documentation should provide details how to do this.

Comment (ehsavoie) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:48]: There is a link to the DataSource subsystem configuration documentation (or am i mistaken in how to use << >>) ?
As it is written it depends on your usage, your database and how many messages and their type (large or not)  you are processing.

Comment (marekkopecky) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc]: > There is a link to the DataSource subsystem configuration documentation (or am i mistaken in how to use << >>) ?

Sorry, you are right, I missed the link. Anyway, you [mentioned some details about timeout, etc.](https://github.com/wildfly/wildfly/pull/13965#discussion_r572024822), which I don't see in that datsource documentation.

>  it depends on your usage, your database and how many messages and their type (large or not) you are processing.

Yes, and the details about this "dependency" needs to be elaborated in the doc. As you said, this RFE is mainly about the documentation and so far the documentation doesn't seems to be providing some insight on details about related configuration.

Comment (marekkopecky) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc]: Ah, looking to latest update and it seems that you already mentioned details about the timeout. But my previous comment is still valid for `the time to return the connection to the pool or renew it`

Comment (ehsavoie) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc]: Should be better now

Comment (marekkopecky) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc]: Thanks, current doc in this PR LGTM, I would just wait for AD approval before my final approve of this PR.

Comment (marekkopecky) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc]: AD is approved, so I'm ok with this doc now as well

**Review by marekkopecky — APPROVED** [review]

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_JDBC_Store_for_Messaging_Journal.adoc:61]: This can be a link that's relative to the current release's docs set. For example:

https://github.com/wildfly/wildfly/blame/master/docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Undertow_using_as_a_Load_Balancer.adoc#L132

Result is at the line above the start of https://docs.wildfly.org/22/Admin_Guide.html#Messaging

**Review by bstansberry — CHANGES_REQUESTED** [review]

**Review by bstansberry — APPROVED** [review]

**Pull request merged**
